Doc/glossary.rst

```diff
@@ -647,10 +648,10 @@ Glossary
       are available a :exc:`StopIteration` exception is raised instead.  At this
       point, the iterator object is exhausted and any further calls to its
       :meth:`__next__` method just raise :exc:`StopIteration` again.  Iterators
-      are required to have an :meth:`__iter__` method that returns the iterator
-      object itself so every iterator is also iterable and may be used in most
-      places where other iterables are accepted.  One notable exception is code
-      which attempts multiple iteration passes.  A container object (such as a
+      are strongly encouraged to have an :meth:`__iter__` method that returns
+      the iterator object itself so an iterator can be used in most places where
+      other iterables are accepted.  One notable exception is code which
+      attempts multiple iteration passes.  A container object (such as a
       :class:`list`) produces a fresh new iterator each time you pass it to the
       :func:`iter` function or use it in a :keyword:`for` loop.  Attempting this
       with an iterator will just return the same exhausted iterator object used
```

Comment:
Surely, if an iterator doesn't have an `__iter__`  method even a single iteration attempt with fail. Further, it's a legitimate technique to use a sequence of iterations over the same iterator, breaking when appropriate, and this won't work without the `__iter__` method. I'm still not understanding the advantage offered by a "partial iterator." Feel free to ignore me if I've missed something obvious.

The this I can't understand is why we should drop the requirement for `__iter__`, thereby allowing people to start writing iterators that aren't iterable (although people have always been able to do so in classes implemented in Python). An iterator that isn't iterable might (IMHO) reasonably be called a "non-iterable iterator". Contraction to "niterator" would give us a term sufficiently close to present-day iterators, and the three-word term is, I feel, definitive.

Just my 2c, and as usual worth what you paid for it.